samples/samples/async_client_sample.py

```diff
@@ -0,0 +1,648 @@
+
+     """Begin Transaction"""
+     transaction_id_blob = transaction_object.begin()
+
+     """Request object for Rollback."""
+     request = RollbackRequest(
+         session = session_object.name,
+         transaction_id = transaction_id_blob,
+     )
+
+     """Rolled back the Transaction."""
+     await async_client.rollback(request = request)
+
+     print("\nEnding task: ", task_num)
+# [END async client rollback]
+
+# [START async client commit]
+async def commit(database_object, async_client, task_num):
+
+     print("\nStarting task: ", task_num)
+
+     # Create the session object associated with the database object
+     session_object = session.Session(database_object)
+
+     """Create the session."""
+     session_object.create()
+
+     """Transaction object associated with the session object"""
+     transaction_object = transaction.Transaction(session_object)
+
+     """Begin Transaction"""
+     transaction_id_blob = transaction_object.begin()
+
+     """Request object for Commit."""
+     request = CommitRequest(
+         transaction_id = transaction_id_blob,
+         session = session_object.name
+     )
+
+     """Commit the transaction"""
+     response = await async_client.commit(request = request)
+
+     """Prints the commit transaction response."""
+     print(response)
+
+     print("\nEnding task: ", task_num)
+# [END async client commit]
+
+# [START async client begin transaction]
+async def begin_transaction(database_object, async_client, task_num):
+
+     print("\nStarting task: ", task_num)
+
+     # Create the session object associated with the database object
+     session_object = session.Session(database_object)
+
+     """Create the session."""
+     session_object.create()
+
+     """Read write transaction object."""
+     transactionOption_object = TransactionOptions(
+         read_write = TransactionOptions.ReadWrite()
+     )
+
+     """Request object for begin transaction."""
+     request = BeginTransactionRequest(
+         session = session_object.name,
+         options = transactionOption_object,
+     )
+
+     """Begins Transaction."""
+     response = await async_client.begin_transaction(request = request)
+
+     """Prints the response."""
+     print(response)
+
+     print("\nEnding task: ", task_num)
+     print("\n")
+# [END async client begin transaction]
+
+# [START async client streaming read]
+async def streaming_read(database_object, async_client, task_num):
+
+     print("\nStarting task: ", task_num)
+
+     # Create the session object associated with the database object
+     session_object = session.Session(database_object)
+
+     """Create the session."""
+     session_object.create()
+
+     """Request object for read."""
+     request = ReadRequest(
+         session = session_object.name,
+         table =  'Singers',
+         columns = ["SingerId", "FirstName", "LastName"],
+         key_set = keyset.KeySet(all_=True)._to_pb(),
```

Comment:
Names with a leading underscore indicate to other programmers that the attribute or method is intended to be private (However, privacy is not enforced in any way). We should avoid calling _to_pb() from here, as it it intended to be private.

Reply:
'key_set' was the required parameter for making this request. And this parameter requires calling _to_pb() method. Hence, I am not sure if there is any way we can avoid calling it.